service/src/main/kotlin/xyz/lebkuchenfm/external/storage/dropbox/DropboxClient.kt

```diff
@@ -162,4 +163,21 @@ class DropboxClient(config: ApplicationConfig) {
         data object ErrorWhenUploadingFile : DropboxClientError()
         data object ErrorWhenCreatingFileUrl : DropboxClientError()
     }
+
+    private fun <T> HeadersBuilder.appendDropboxApiArg(value: T, serializer: KSerializer<T>) {
+        // https://www.dropbox.com/developers/reference/json-encoding
+        fun String.encodedForDropbox() = buildString {
+            for (char in this@encodedForDropbox) {
+                if (char.code in 32..<127) {
```

Comment:
Since the root of the problem is passing stuff in headers, what we actually need is just URL encoding like this:
```
URLEncoder.encode(value, Charsets.UTF_8)
```
That's also what the documentations says:
> If you use the arg URL parameter, generate a JSON string, encode it as UTF-8, then use the application/x-www-form-urlencoded rules to encode the value.

Reply:
I tried that. But then dropbox responded with http 400 and info that it expected `Content-Type: application/octet-stream`

Reply:
nvm, must have had a stroke yesterday or sth.

done